src/input.rs

```diff
@@ -16,83 +21,80 @@ pub(crate) struct App {
 }
 
 impl App {
-    fn stdin_replace(&self, is_tty: bool) -> Result<()> {
-        let mut buffer = Vec::with_capacity(256);
-        let stdin = std::io::stdin();
-        let mut handle = stdin.lock();
-        handle.read_to_end(&mut buffer)?;
-
-        let stdout = std::io::stdout();
-        let mut handle = stdout.lock();
-
-        handle.write_all(&if is_tty {
-            self.replacer.replace_preview(&buffer)
-        } else {
-            self.replacer.replace(&buffer)
-        })?;
-
-        Ok(())
-    }
-
     pub(crate) fn new(source: Source, replacer: Replacer) -> Self {
         Self { source, replacer }
     }
+
     pub(crate) fn run(&self, preview: bool) -> Result<()> {
-        let is_tty = std::io::stdout().is_terminal();
+        let sources: Vec<(PathBuf, Mmap)> = match &self.source {
+            Source::Stdin => {
+				let mut handle = stdin().lock();
+				let mut buf = Vec::new();
+				handle.read_to_end(&mut buf)?;
+				let mut mmap = MmapOptions::new()
+					.len(buf.len())
+					.map_anon()?;
+				mmap.copy_from_slice(&buf);
+				let mmap = mmap.make_read_only()?;
+				vec![(PathBuf::from("STDIN"), mmap)]
```

Comment:
Using `STDIN` as a sigil for `stdin` here can cause other issues to slip in. I think it's distinct enough to merit its own variant. Maybe something like

```rust
enum Source {
    Stdin,
    Path(PathBuf),
}
```

You can also `impl Display` for it to keep using it for formatting when displaying results

Reply:
There's actually `pub(crate) enum Source`, but it holds a `Vec<PathBuf>` in the `::Files` variant. I'm not sure as to reuse it or introduce a new `PathBuf` variant or…

Reply:
Ah I forgot about that. Maybe a name like `InputSource` or something then? This is distinct enough from `Source` that it should be a separate thing

Reply:
Instead of introducing a duplicate enum I think we can move the `vec![Stdin]` part forward to `::main::try_main()`, passing a `Vec<Source::{Stdin, File}>` to `App` rather than `Source::{Stdin, Files(Vec<PathBuf>)}`.

Reply:
Sure, that seems like a fine way to handle it